### PR TITLE
[Merged by Bors] - feat(GroupTheory/OrderOfElement): If `a ≤ g • a`, then `g • a = a`

### PR DESCRIPTION
**Pull request by tb65536**
This PR adds a few lemmas about group actions on partial orders.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/Action/Basic.lean [hunk 1]
```diff
@@ -146,3 +147,19 @@ variable [Group α] [Monoid β] [MulAction α β] [SMulCommClass α β β] [IsSc
   ⟨fun h => inv_smul_smul g m ▸ h.smul g⁻¹, IsUnit.smul g⟩
 
 end SMul
+
+lemma le_pow_smul {G : Type*} [Monoid G] {α : Type*} [Preorder α] {g : G} {a : α}
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Group/Action/Basic.lean:151]: I think this should go somewhere under `Mathlib.Algebra.Order.Group`; maybe in a new file. CC @YaelDillies for suggestions

Comment (YaelDillies) [reply Mathlib/Algebra/Group/Action/Basic.lean:151]: Agreed. `Algebra.Group.something` should be order-free.

Comment (tb65536) [reply Mathlib/Algebra/Group/Action/Basic.lean]: Good point. I've moved them to `Algebra.Order.Group.Action`. Still had to add an import, but having `Algebra.Order.Group.Action` import `Algebra.Group.Action.Defs` doesn't seem so bad.

Comment (tb65536) [reply Mathlib/Algebra/Group/Action/Basic.lean]: But let me know if there's a better home.

File: Mathlib/GroupTheory/OrderOfElement.lean [hunk 2]
```diff
@@ -962,6 +963,22 @@ theorem pow_gcd_card_eq_one_iff : x ^ n = 1 ↔ x ^ gcd n (Fintype.card G) = 1 :
     let ⟨m, hm⟩ := gcd_dvd_left n (Fintype.card G)
     rw [hm, pow_mul, h, one_pow]⟩
 
+lemma smul_eq_of_le_smul
+    {G : Type*} [Group G] [Finite G] {α : Type*} [PartialOrder α] {g : G} {a : α}
+    [MulAction G α] [CovariantClass G α HSMul.hSMul LE.le] (h : a ≤ g • a) : g • a = a := by
```
Comment (YaelDillies) [inline Mathlib/GroupTheory/OrderOfElement.lean:968]: there i now an abbrev for this
```suggestion
    [MulAction G α] [SMulLeftMono G α] (h : a ≤ g • a) : g • a = a := by
```
or something

Comment (tb65536) [reply Mathlib/GroupTheory/OrderOfElement.lean:984]: We have `PosSMulMono`, but I can't seem to find `SMulMono`.

**Review by YaelDillies — APPROVED** [review]
Thanks! 🚀

maintainer merge

**Pull request closed without merging**